### PR TITLE
Exclude party in the grouping when is None or 0

### PR DESCRIPTION
**Pull request by raimonesteve**
#043720

### CODE REVIEW

File: stock.py [hunk 1]
```diff
@@ -70,8 +70,10 @@ class Lot(metaclass=PoolMeta):
     @classmethod
     def _get_quantity(cls, records, name, location_ids,
             grouping=('product',), grouping_filter=None, position=-1):
-        if 'party' in Transaction().context:
-            party_id = Transaction().context.get('party')
+        context = Transaction().context
+        party_id = context.get('party')
+        # exclude party grouping when party is None o 0
+        if party_id:
```
Comment (albertca) [inline stock.py]: The problem with this change is that if we don't group by party, when the move has no party, the quantity of lots from other parties will also be shown and that is not what we want.

If party is empty in the stock move, then we want only to show the quantity available for lots on moves with no party.

Comment (raimonesteve) [reply stock.py]: Ok. I understand and I found that problem is the grouping_filter [None], the query is: 

```
AND ("b"."party" IN (NULL))
```

Solved to replace _("b"."party" IN (NULL))_ to  _("b"."party" IS NULL)_

Example:

```
 location | product | lot  | party | quantity 
----------+---------+------+-------+----------
       22 |     586 | 8192 |       |      100
       22 |     586 | 8265 |       |     2492
```

Working progress.....

Comment (raimonesteve) [reply stock.py]: La part del lot i mostrar les quantitats, si té party o és None, quedaria solventada amb: 

```
grouping_filter = grouping_filter + ([], [party_id] if party_id else None,)
```

Ara bé, seguint el mateix patró de debug, no hem quadra en el cas del producte.quantity. Aquí també si en el context tenim "party" hem d'excloure'ls. Per ara he sobrescrit el _get_quantity i si tenim party en el context, excloure aquells moviments que no siguin del tercer per saber la quantitat.

He afegit en el test pq s'entengui, però si li passem party en el context, el product.quantity ens retorna 0 i no el valor que ens esperem (el que s'ha marcat en el assertEqual). S'ha mirat i remirat, i no trobo motiu i la query la veig bé.

Caldria confirmar aquest comportament i si que pot ser la causa.

```
            with transaction.set_context(locations=[storage.id]):
                product = Product(product.id)
                self.assertEqual(party.quantity, 20.0)

            with transaction.set_context(locations=[storage.id], party=customer.id):
                product = Product(product.id)
                self.assertEqual(party.quantity, 5.0)

            with transaction.set_context(locations=[storage.id], party=None):
                product = Product(product.id)
                self.assertEqual(party.quantity, 15.0)
```

**Review by albertca — APPROVED** [review]

**Comment by raimonesteve** [comment]
Depreacted!

**Pull request closed without merging**